source/rust_verify/src/rust_to_vir_func.rs

```diff
@@ -470,14 +470,21 @@ pub(crate) fn check_item_fn<'tcx>(
             );
         }
 
-        let typ = mid_ty_to_vir(
-            ctxt.tcx,
-            &ctxt.verus_items,
-            id,
-            span,
-            is_ref_mut.map(|(t, _)| t).unwrap_or(input),
-            false,
-        )?;
+        let typ = {
+            let typ = mid_ty_to_vir(
+                ctxt.tcx,
+                &ctxt.verus_items,
+                id,
+                span,
+                is_ref_mut.map(|(t, _)| t).unwrap_or(input),
```

Comment:
This strips the decoration if `is_ref_mut.is_some()`. The proposed change keeps track of it in the return value for `is_mut_ty` it and re-introduces it as a decoration of the non-`&mut` type.